hyp3lib/get_orb.py

```diff
@@ -1,197 +1,147 @@
-
-  start_time = start_time - sec60
-  end_time = end_time + sec60
-
-  q = precise+'validity_stop__gt='+start_time.strftime('%Y-%m-%dT%H:%M:%S')+'&validity_start__lt='+end_time.strftime('%Y-%m-%dT%H:%M:%S')
-  orbitFile,url = getPageContentsESA(q, False)
-  if url:
-    print("Using url {}".format(url))
-  else:
-    print("Unable to find POEORB - Looking for RESORB")
-    q=restituted+'validity_stop__gt='+start_time.strftime('%Y-%m-%dT%H:%M:%S')+'&validity_start__lt='+end_time.strftime('%Y-%m-%dT%H:%M:%S')
-    orbitFile,url = getPageContentsESA(q, False)
-    if url:
-      print("Using url {}".format(url))
-    else:
-      error = 'Could not find orbit file on ESA website'
-      raise FileException(error)
-
-  return url, orbitFile
-
-
-def fetchOrbitFile(urlOrb,stateVecFile,verify):
-  hostname = urlparse(urlOrb).hostname
-  session = requests.Session()
-  session.mount(hostname, HTTPAdapter(max_retries=10))
-  request = session.get(urlOrb, timeout=60, verify=verify)
-  f = open(stateVecFile, 'w')
-  f.write(request.text)
-  f.close()
-
-
-def downloadSentinelOrbitFileProvider(granule, provider, directory):
-  if provider.upper() == 'ASF':
-    urlOrb, fileNameOrb = getOrbFile(granule)
-    verify = True
-  elif provider.upper() == 'ESA':
-    urlOrb, fileNameOrb = getOrbitFileESA(granule)
-    verify = False
-
-  if directory:
-    stateVecFile = os.path.join(directory, fileNameOrb)
-  else:
-    stateVecFile = fileNameOrb
-
-  if not os.path.isfile(stateVecFile):
-    if len(stateVecFile) > 0:
-      fetchOrbitFile(urlOrb,stateVecFile,verify)
-      return stateVecFile
-    else:
-      return None
-  else:
-    print("Using existing orbit file; provider unknown")
-    return stateVecFile
-
-
-def downloadSentinelOrbitFile(granule,directory=None):
+    best = None
+    for file in file_list:
+        file = file.strip()
+        t = re.split('_', file.replace('T', '').replace('V', ''))
+        if len(t) > 7:
+            start = t[6]
+            end = t[7].replace('.EOF', '')
+            if start < timestamp < end:
+                d = ((int(timestamp) - int(start)) + (int(end) - int(timestamp))) / 2
+                if d > d1:
+                    best = file
+                    d1 = d
+    return search_url + best
+
+
+def get_orbit_url(granule: str, orbit_type: str = 'AUX_POEORB', provider: str = 'ESA'):
+    """Get the URL of an orbit file from a provider
+
+    Args:
+        granule: Granule name to find an orbit file for
+        orbit_type: Orbit type to download
+        provider: Povider name to download the orbit file from
+
+    Returns:
+        orbit_url: The url to the matched orbit file
+    """
+    platform = granule[0:3]
+    time_stamps = re.split('_+', granule)[4:6]
+
+    if provider.upper() == 'ESA':
+        delta = timedelta(seconds=60)
+        start_time = datetime.strptime(time_stamps[0], '%Y%m%dT%H%M%S') - delta
+        end_time = datetime.strptime(time_stamps[1], '%Y%m%dT%H%M%S') + delta
```

Comment:
wanted to revisit start/endtimes to match the sentinel-1 toolbox code at https://github.com/senbox-org/s1tbx/blob/7720f3ada12f7ea0394ae2c8d180fad270494204/s1tbx-io-ephemeris/src/main/java/org/esa/s1tbx/io/orbits/sentinel1/SentinelPODOrbitFile.java#L157
```java
validity_stop__gt=date+"T23:59:59";
validity_start__lt=date+"T"+hour+":"+minute+":"+second;
```
I *think* `date` is the acquisition start time, I can dig to confirm that.

Reply:
Going to punt this to future work